crates/re_query/src/entity_view.rs

```diff
@@ -107,10 +107,11 @@ impl ComponentWithInstances {
             0
         } else {
             // Otherwise binary search to find the offset of the instance
-            keys.binary_search(&instance_key.0).ok()? as u32
+            keys.binary_search(&instance_key.0).ok()?
         };
 
-        Some(self.values.as_arrow_ref().slice(offset as _, 1))
+        let arrow_ref = self.values.as_arrow_ref();
+        (arrow_ref.len() > offset).then(|| arrow_ref.slice(offset, 1))
```

Comment:
This is a very corrupt/illegal state that the store is in; the Rust SDK shouldn't have allowed it in the first place.
But it can still happen anyway if a user decides to upload data via the ABI, so we need to handle it anyhow...

This requires an `error_once!` at the very least

Reply:
it's embarrassingly simple to get into that state. As long as we don't have anything avoiding that I'm against logging an error here

Reply:
resolved via  #2059